ifrs17/Import/Importers.ipynb

```diff
@@ -485,19 +507,42 @@
         {
             "cell_type": "code",
             "source": [
-                "public async Task<ImportArgs[]> GetAllArgsAsync(ImportArgs primaryArgs)",
+                "public async Task<ImportArgs[]> GetAllArgsAsync(ImportArgs args, IDataSource dataSource, string format)",
                 "\n{",
-                "\n    if (primaryArgs.Scenario != null)",
-                "\n        return primaryArgs.RepeatOnce().ToArray();",
-                "\n",
-                "\n    var secondaryArgs = await DataSource.Query<PartitionByReportingNodeAndPeriod>() ",
-                "\n                               .Where(x => x.ReportingNode == primaryArgs.ReportingNode && x.Year == primaryArgs.Year && x.Month == primaryArgs.Month && x.Scenario != null)",
-                "\n                               .Select(x => new ImportArgs(x.ReportingNode, x.Year, x.Month, default(Periodicity), x.Scenario, ImportFormats.Cashflow)).ToArrayAsync();",
+                "\n    ImportArgs[] allArgs;",
+                "\n    switch(format)",
+                "\n    {",
+                "\n        case ImportFormats.YieldCurve : ",
+                "\n        {",
+                "\n            if(args.Scenario == null) {",
+                "\n                var scenariosWithYieldCurves = await dataSource.Query<YieldCurve>().Where(x => x.Year == args.Year && x.Month == args.Month && x.Scenario != null)",
+                "\n                    .Select(x => x.Scenario).Distinct().ToArrayAsync();",
+                "\n                var targetPartitions = await dataSource.Query<PartitionByReportingNodeAndPeriod>()",
+                "\n                    .Where(x => x.Year == args.Year && x.Month == args.Month && !scenariosWithYieldCurves.Contains(x.Scenario)).OrderBy(x => x.Scenario).ToArrayAsync();",
+                "\n                var targetScenarios = targetPartitions.Where(x => x.Scenario != null).Select(x => x.Scenario);",
+                "\n                if(targetScenarios.Any()) ApplicationMessage.Log(Warning.ScenarioReCalculations, String.Join(\", \", targetScenarios));",
+                "\n                allArgs = targetPartitions.Select(x => new ImportArgs(x.ReportingNode, x.Year, x.Month, default(Periodicity), x.Scenario, ImportFormats.Cashflow)).ToArray();",
```

Comment:
don't we need to add args here?

Reply:
No because for this case the args (taken from the Main tab) is not usable. It doesn't contain the ReportingNode and only serves to calculate all relevant args to be computed.